### DISCUSSION TITLE
Report about downloading images and books with an isbn13 starting with 979.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#268] [Ideas]
Central Library Book Delivery Statistics Report 2022-12
Total number of books  : 592
Book with an isbn13 starting with 979 : 47 (or 8%)

- no cover with Amazon feature because does not support books with isbn13 starting with 979

if the isbn-13 are transformed into isbn-10 for the 979 (theoretically impossible in a "book" context but BQ agrees to do it) then a cover of another book appears, which is not very useful for the user.

books concerned 2022-12 by isbn13 -> isbn10 = bad cover
979-1090-2130-29
979-1090-7430-07
979-1090-7436-63
979-1090-7439-77
979-1092-3047-56
979-1093-2952-82
979-1095-0302-32
979-1095-7188-33
979-1096-3101-35
979-1096-3159-25

-------------------------------------------------------
Central Library Book Delivery Statistics Report 2022-07
Total number of books  : 417
Book with an isbn13 starting with 979 : 40 (9 or 10%)

books concerned 2022-07 by isbn13 -> isbn10 = bad cover
979-1032-4048-98
979-1096-6882-34
979-1092-3049-30
979-1092-3049-23
979-1092-7525-71
979-1096-4052-44
979-1034-7314-59
979-1034-7377-41
979-1092-7631-88

-------------------------------------------------------
EXCEPTIONAL same cover with 2 different isbn10 :-)
979-1092-3049-23
979-1092-3047-56

Ideas of evolutions
- Block the transformation of isbn-13 starting with 979 to isbn-10 (this will avoid bad surprises for the covers)

Another coverage search engine is available in BQ, it is based on the alternative_id_1 field, but you have to do everything manually :

- Modify the Unimarc SRU parser to take this line into account and place the ark:/12148/cb43651335p information in the alternate_id_1 field
<mxc:controlfield tag="003">http://catalogue.bnf.fr/ark:/12148/cb43651335p</mxc:controlfield> 

not in the form of a ticket because not urgent, it's just a report and ideas for the TODO list

**Comment by textbrowser** [comment]
One completed. :P

**Reply by meteos77** [reply]
super I will test tomorrow

I saw that you also gave the possibility to rename the personal sql queries, it's cool, I will be able to improve my list from the BQ interface

**Reply by meteos77** [reply]
Can you explain this note?   _if a future database does not have a database driver_

**Comment by textbrowser** [comment]
A future database is one which will be accessed in the future. For example, the current database's driver is SQLite and the database to be opened has a PostgreSQL driver which is not available on the system. I suppose the database could be inspected during initialization. They're inspected whenever they're accessed.

**Comment by textbrowser** [comment]
2nd completed.

https://github.com/textbrowser/biblioteq/commit/8f7012ffe1dc00858d4b942fe486da05e80033d3

**Comment by textbrowser** [comment]
Discussion closed.
